ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueConsumption.scala

```diff
@@ -243,6 +262,86 @@ object KeyValueConsumption {
     )
   }
 
+  private[kvutils] def outOfTimeBoundsEntryToUpdate(
+      recordTime: Timestamp,
+      outOfTimeBoundsEntry: DamlOutOfTimeBoundsEntry): Option[Update] = {
+    val wrappedLogEntry = outOfTimeBoundsEntry.getEntry
+    val duplicateUntilMaybe = parseOptionalTimestamp(
+      outOfTimeBoundsEntry.hasDuplicateUntil,
+      outOfTimeBoundsEntry.getDuplicateUntil)
+    val tooEarlyUntilMaybe = parseOptionalTimestamp(
+      outOfTimeBoundsEntry.hasTooEarlyUntil,
+      outOfTimeBoundsEntry.getTooEarlyUntil)
+    val tooLateFromMaybe = parseOptionalTimestamp(
+      outOfTimeBoundsEntry.hasTooLateFrom,
+      outOfTimeBoundsEntry.getTooLateFrom)
+    val deduplicated = duplicateUntilMaybe.exists(recordTime <= _)
+    val tooEarly = tooEarlyUntilMaybe.exists(recordTime <= _)
+    val tooLate = tooLateFromMaybe.exists(recordTime > _)
+    val invalidLedgerTime = tooEarly || tooLate
```

Comment:
`invalidLedgerTime` is a bit too specific of a name, as ledger time is only defined for transaction submissions. Maybe rename this to `invalidRecordTime`?

Reply:
Renamed.